extension/solarwindsapmsettingsextension/config.go

```diff
@@ -2,40 +2,10 @@
 // SPDX-License-Identifier: Apache-2.0
 
 package solarwindsapmsettingsextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/solarwindsapmsettingsextension"
-
-import (
-	"errors"
-	"strconv"
-	"strings"
-	"time"
-)
+import "time"
 
 type Config struct {
-	Endpoint string `mapstructure:"endpoint"`
-	Key      string `mapstructure:"key"`
-	Interval string `mapstructure:"interval"`
-}
-
-func (cfg *Config) Validate() error {
```

Comment:
Why is validation the removed?

Reply:
Returning an error in `Config.Validate()` will cause collector not starting up. 

While we are testing with the `opentelemetry-lambda` collector lambda layer, the collector error from this function can cause the lambda function (user's lambda function) timeout. i.e. A bad config in this extension can break user's lambda function. 

While we think a bad config in this extension (e.g. a wrong endpoint causing `no such host` gRPC error or a missing / wrong key to get a setting should not be a fatal to the collector process. 

So we moved the logic to the extension.

When there is a bad config, the extension will log messages and will be in noop.

Reply:
Done. Added back `otelcol validate` function